### ISSUE TITLE
Different classification scores when using multi_class_prob

### ISSUE THREAD

**Issue by yosipk** [flairNLP/flair#2376]
**Describe the bug**
When using the multi-class classifier's `predict` method with `multi_class_prob` argument set to `True` I would expect that the model behaves the same as with this flag set to `False` except that instead of probabilities for all classes it writes into the predicted labels only the label with highest score. However, that is not the case. 

In [L493](https://github.com/flairNLP/flair/blob/288f7ce4a0680ac8b3ab27d252f217d2bbd5a73c/flair/nn/model.py#L493) we check if the `multi_class_prob` is set, and if it is the scores are computed by pulling logits through the sigmoid in [L494](https://github.com/flairNLP/flair/blob/288f7ce4a0680ac8b3ab27d252f217d2bbd5a73c/flair/nn/model.py#L494), while if `multi_class_prob` is not set the scores are computed by pulling logits through the softmax in [L507](https://github.com/flairNLP/flair/blob/288f7ce4a0680ac8b3ab27d252f217d2bbd5a73c/flair/nn/model.py#L507). 

**To Reproduce**
Run the model that inherits from `DefaultClassifier` with and without `multi_class_prob` set and compare the scores for the highest scoring class. They scores are different. The sum of scores with `multi_class_prob` is not 1. 

**Expected behavior**
The scores should be the same. The sum of scores with `multi_class_prob` should be 1 (up to numerical error).

**Environment (please complete the following information):**
 - OS: MacOS (20.5.0 Darwin Kernel)
 - Version: 0.8.1

**Comment by yosipk** [comment]
The code
```python
from flair.models import TextClassifier
from flair.data import Sentence

# load tagger
classifier = TextClassifier.load('sentiment')

sentence = Sentence("enormously entertaining for moviegoers of any age.")
classifier.predict(sentence, multi_class_prob=False)
print(sentence)

sentence = Sentence("enormously entertaining for moviegoers of any age.")
classifier.predict(sentence, multi_class_prob=True)
print(sentence)
print(sum([label.score for label in sentence.labels]))
```

Output with current main/master branch:
```
Sentence: "enormously entertaining for moviegoers of any age ."   [− Tokens: 8  − Sentence-Labels: {'label': [POSITIVE (0.9976)]}]
Sentence: "enormously entertaining for moviegoers of any age ."   [− Tokens: 8  − Sentence-Labels: {'label': [NEGATIVE (0.0546), POSITIVE (0.9597)]}]
1.0142793580889702
```

Output with linked PR:
```
Sentence: "enormously entertaining for moviegoers of any age ."   [− Tokens: 8  − Sentence-Labels: {'label': [POSITIVE (0.9976)]}]
Sentence: "enormously entertaining for moviegoers of any age ."   [− Tokens: 8  − Sentence-Labels: {'label': [NEGATIVE (0.0024), POSITIVE (0.9976)]}]
1.000000075204298
```
